src/opening-period/page/CreateNewOpeningPeriodPage.tsx

```diff
@@ -9,37 +9,84 @@ import { ResourceInfo } from '../../resource/page/ResourcePage';
 import Datepicker from '../../components/datepicker/Datepicker';
 import { ErrorToast, SuccessToast } from '../../components/notification/Toast';
 import './CreateNewOpeningPeriodPage.scss';
-import { Resource, ResourceState } from '../../common/lib/types';
+import {
+  TimeSpan as TimeSpanApiFormat,
+  TimeSpanFormFormat,
+  Resource,
+  ResourceStateOption,
+} from '../../common/lib/types';
 import {
   dateApiFormat,
   dateFormFormat,
 } from '../../common/utils/date-time/format';
+import Button from '../../components/button/Button';
+import OpeningPeriodDescription from '../description/OpeningPeriodDescription';
+
+import TimeSpan from '../time-span/TimeSpan';
 
 type Inputs = {
   openingPeriodTitle: string;
   openingPeriodOptionalDescription: string;
   openingPeriodBeginDate: string;
   openingPeriodEndDate: string;
+  timeSpans: Array<TimeSpanFormFormat>;
 };
 
 type SubmitStatus = 'init' | 'succeeded' | 'error';
 
+function formatTimeSpansToApiFormat(
+  timeSpans: TimeSpanFormFormat[]
+): TimeSpanApiFormat[] {
+  return timeSpans.map((timeSpan) => {
+    return {
+      description: {
+        fi: timeSpan.description,
+        sv: null,
+        en: null,
+      },
+      end_time: `${timeSpan.endTime}:00`,
+      start_time: `${timeSpan.startTime}:00`,
+      resource_state: timeSpan.resourceState,
+      weekdays: timeSpan.weekdays.reduce(
```

Comment:
JS native map has index as a second argument in the callback function. Would be easier to use that in the render function? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
Js native map will probably create as many elements as there are in the original array? In the api format the array length is reduced; it consists of only those days that are selected

Reply:
This is resolved then